edxval/api.py

```diff
@@ -296,11 +341,36 @@ def get_video_info_for_course_and_profile(course_id, profile_name):
         raise ValInternalError(error_message)
 
     # DRF serializers were causing extra queries for some reason...
-    return {
-        enc_vid.video.edx_video_id: {
-            "url": enc_vid.url,
-            "file_size": enc_vid.file_size,
-            "duration": enc_vid.video.duration,
-        }
-        for enc_vid in encoded_videos
-    }
+    return_dict = {}
+    for enc_vid in encoded_videos:
+        return_dict.setdefault(enc_vid.video.edx_video_id, {}).update(
+            {
+                "duration": enc_vid.video.duration,
+                enc_vid.profile.profile_name: {
```

Comment:
If someone creates a profile with the name "duration", this will break. As per the original comment, I think it should be like the following:

```
{
    "video_id_1": {
        "duration": 100,
        "encodings": {
            "profile1": {
                "url": "http://example.com/foo.mp4",
                "file_size": 555,
            },
            ...
        }
    },
    ...
}
```